.github/workflows/reusable-native-tests.yml

```diff
@@ -13,10 +13,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+      - id: read-java
+        run: echo "version=$(cat .java-version)" >> "$GITHUB_OUTPUT"
       - uses: graalvm/setup-graalvm@2f25c0caae5b220866f732832d5e3e29ff493338 # v1.2.1.1
         with:
           version: "latest"
-          java-version: "21"
+          java-version: "${{ steps.read-java.outputs.version }}"
```

Comment:
This can be potentially once https://github.com/graalvm/setup-graalvm/pull/95 is done :)